Enable creation of bindless images backed by host USM

przemektmalon opened the pull request:
Small patch to enable bindless images backed by host USM in the CUDA adapter.

Host and Device USM pointers are usable across the host and device for all versions of CUDA that we support. There is no need to provide the `CU_MEMHOSTALLOC_DEVICEMAP` flag during allocation, or calling `cuMemHostGetDevicePointer` to retrieve a device usable address.

Passing a `CU_MEMHOSTALLOC_WRITECOMBINED` flag to the host USM allocation will enhance performance in certain scenarios, however, an extension allowing this is not yet available.

Related DPC++ PR: https://github.com/intel/llvm/pull/16607

kbenzie commented:
I've removed *ready to merge* as https://github.com/intel/llvm/pull/16607 still requires approvals (except from *unified-runtime-reviewers*) and must have all checks passing. Please add the lable again once these conditions are satisfied.

przemektmalon commented:
Added ready to merge label. 
https://github.com/intel/llvm/pull/16607 passed CI and has approval.
I've just re-based this PR and updated the tag in LLVM, so CI is re-running.
